extractor/src/main/scala/com/digitalasset/extractor/Extractor.scala

```diff
@@ -104,23 +113,35 @@ class Extractor[T <: Target](config: ExtractorConfig, target: T) {
     system.terminate().map(_ => killSwitch.shutdown())
   }
 
-  private def fetchPackages(client: LedgerClient, writer: Writer): Future[Unit] = {
+  private def fetchPackages(client: LedgerClient, writer: Writer): Future[PackageStore] = {
     for {
-      packageStore <- LedgerReader.createPackageStore(client)
-      _ <- writer.handlePackages(packageStore.valueOr(error => throw new RuntimeException(error)))
-    } yield ()
+      packageStoreE <- LedgerReader.createPackageStore(client): Future[String \/ PackageStore]
+      packageStore <- toFuture(packageStoreE): Future[PackageStore]
+      _ <- writer.handlePackages(packageStore)
+    } yield packageStore
   }
 
-  private def selectTransactions: TransactionFilter = {
+  private def selectTransactions(parties: ExtractorConfig.Parties): TransactionFilter = {
```

Comment:
The `parties` argument is unused so should be removed. If you want to keep the argument, move the function to the companion.

Reply:
my bad, I was actually going to use it :)